nni/experiment/config/experiment_config.py

```diff
@@ -141,7 +141,7 @@ def _canonicalize(self, _parents):
                 msg = f'nni_manager_ip is not set, please make sure {ip} is accessible from training machines'
                 logging.getLogger('nni.experiment.config').warning(msg)
 
-    def _validate_canonical(self):
+    def _validate_canonical(self, validate_tuner: bool = True): # FIXME: remove validate_tuner
```

Comment:
Don't violate base class' signature.
If you need a temp hack, use `type(self)`.

Reply:
then this base class should import its subclass?

Reply:
Use `type(self).__name__` or delayed import.
It's a hack, and hack should be as "local" as possible.

Reply:
too me, it is more urgly, because base class has the logic of sub-class. anyway, updated